services/viva-ms/lambdas/syncOfficers.js

```diff
@@ -0,0 +1,160 @@
+    return null;
+  }
+
+  // Convert DynamoDB case data to plain object
+  const unMarshalledCaseData = dynamoDbConverter.unmarshall(event.detail.dynamodb.NewImage);
+
+  const { details } = unMarshalledCaseData;
+  if (!('administrators' in details)) {
+    details['administrators'] = [];
+  }
+  const { administrators } = details;
+
+  // Get Viva applicant officer(s)
+  const [vadaMyPagesError, vadaMyPagesResponse] = await to(
+    sendVadaMyPagesRequest(unMarshalledCaseData)
+  );
+  if (vadaMyPagesError) {
+    return console.error('(Viva-ms) syncOfficers', vadaMyPagesError);
+  }
+
+  const { officer } = vadaMyPagesResponse;
+  const vivaAdministrators = parseVivaOfficers(officer);
+
+  if (deepEqual(vivaAdministrators, administrators)) {
+    return null;
+  }
+
+  // Out of sync. Update
+  const TableName = config.cases.tableName;
+  const { PK, SK } = unMarshalledCaseData;
+
+  const UpdateExpression = 'SET details.administrators = :newAdministrators';
+  const ExpressionAttributeValues = { ':newAdministrators': vivaAdministrators };
+
+  const dynamoDbParams = {
+    TableName,
+    Key: {
+      PK,
+      SK,
+    },
+    UpdateExpression,
+    ExpressionAttributeValues,
+    ReturnValues: 'UPDATED_NEW',
+  };
+
+  const [updateError] = await to(sendUpdateRequest(dynamoDbParams));
+  if (updateError) {
+    return console.error('(Viva-ms) syncOfficers', updateError);
+  }
+
+  return true;
+};
+
+/**
+ * Handler sending GET call to the Viva API adapter mypages endpoint
+ *
+ * @param {object} caseData
+ * @return {object} Viva applicant officer(s)
+ */
+async function sendVadaMyPagesRequest(caseData) {
+  const { PK } = caseData;
+  const personalNumber = PK.substring(5);
+  const ssmParams = await SSMParams;
+
+  const { hashSalt, hashSaltLength } = ssmParams;
+  const personalNumberEncoded = hash.encode(personalNumber, hashSalt, hashSaltLength);
+
+  const { vadaUrl, xApiKeyToken } = ssmParams;
+  const requestClient = request.requestClient({}, { 'x-api-key': xApiKeyToken });
+
+  const vadaMyPagesUrl = `${vadaUrl}/mypages/${personalNumberEncoded}`;
+
+  const [error, vadaMyPagesResponse] = await to(
+    request.call(requestClient, 'get', vadaMyPagesUrl, null)
+  );
+
+  if (error) {
+    if (error.response) {
+      // The request was made and the server responded with a
+      // status code that falls out of the range of 2xx
+      throwError(error.response.status, error.response.data.message);
+    } else if (error.request) {
+      // The request was made but no response was received
+      // `error.request` is an instance of http.ClientRequest in node.js
+      throwError(500, error.request.message);
+    } else {
+      // Something happened in setting up the request that triggered an Error
+      throwError(500, error.message);
+    }
+  }
+
+  return vadaMyPagesResponse.data.person.cases.vivacases.vivacase.officers;
+}
+
+/**
+ * Handler updateing data stored in DynamoDB
```

Comment:
Grammatic error, updateing => updating

Reply:
Also what data do we update? Is this a specific function for the case table in dynamodb or is it a general update function for a dynamo table :)